---
title: "Discussion"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Conclusions:

1. Aligners

-comparisons (explain differences and similarities between them)

After running the data through the varying computer models, we found that the combination of the differential expression programs and the aligners used plays an important role in determining which genes were up- and down-regulated. Because the identification of differentially-expressed genes are highly dependent on the transcriptomics pipeline, this proves that RNA-Seq techniques differ and it is essential to try to figure out the differences and similarities between each pipeline and DE programs and why we would use a certain one over another. 

Looking at the MA plots was helpful in determining the differences between the aligners used. We found that  
The differences in the results of these aligners may be due to their methodology when they are mapping gene sequences. Both Bowtie2 and Sailfish are splice-aware aligners (method = slow), so the sequence reads were mapped directly to the reference genome. Salmon and Kallisto, on the other hand, are pseudo-aligners (method = fast), meaning that the reads are not directly aligned. Instead, they are determined based on their compatability with known transcripts. 

[]Bowtie
[]Kallisto
[]Salmon
[]Sailfish


2. Differential Expression Programs

*there should be more differences between these than between the aligners*

-are there any consistent patterns in the aligners? (look at Ven Diagrams)
-look at conservativeness between aligners

[]Limma
[]DESeq
[]EdgeR
