---
title: "Discussion"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Conclusions:

1. Aligners

-comparisons (explain differences and similarities between them)

After running the data through the varying computer models, we found that the combination of the differential expression programs and the aligners used plays an important role in determining which genes were recognized as up- and down-regulated. Because the identification of differentially-expressed genes are highly dependent on the transcriptomics pipeline, this proves that RNA-Seq techniques differ and it is essential to try to figure out the differences and similarities between each pipeline and DE programs and why we would use a certain one over another. 
fsd
Looking at the MA plots was helpful in determining the differences between the aligners used. If genes were shown to be further towards the 1 and -1 on the y-axis, this meant that they were more differentially expressed. The MA plot results tells us what the relative sensitivity of the differential expression programs. 

The differences in the results of these aligners may be due to their methodology when they are mapping gene sequences. Both Bowtie2 and Sailfish are splice-aware aligners (method = slow), so the sequence reads were mapped directly to the reference genome. Salmon and Kallisto, on the other hand, are pseudo-aligners (method = fast), meaning that the reads are not directly aligned. Instead, they are determined based on their compatability with known transcripts. 

MA plots: results should tell you what the relative sensitivity of the program
each of the points represents one of the transcripts

a = abundance of transcript (counts per million)
genes on left = low abundance
genes on right 

m = fold change between the two samples
if a sample is on the top half, it is more highly expressed in the RNAi treated
if on the bo

color dots indicate genes that met the cut off, significiantly different
more dots = more sensitive, 

challenge:


[]Bowtie
[]Kallisto
[]Salmon

All genes with this aligner were expressed at lower levels than compared to the other aligners. But, they were also spread out more across the x-axis. 

[]Sailfish

look at venn 
Looking at the venn diagrams, it is clear that there were certain combinations that did not work. For example, Salmon and DESeq2 was not very successful in recognizing the expression of the genes in the sample. 

2. Differential Expression Programs

*there should be more differences between these than between the aligners*

-are there any consistent patterns in the aligners? (look at Venn Diagrams)
-look at conservativeness between aligners (which is identifying the fewest genes) how do you tell what is too conservative or not 
for bowtie 2, limma was the most conservative, followed by DESeq2, then EdgeR
(numbers = number of differential genes )

either certain aligners don't play well with certain programs, OR someone made a mistake

[]Limma
for both limma and deseq2, the salmon aligner is throwing everything off 
[]DESeq
it really matters what aligner you use for DESeq2
[]EdgeR
