---
title: "Discussion"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

#Conclusions:

1. Aligners

After running the data through the varying computer models, we found that the combination of the differential expression programs and the aligners used plays an important role in determining which genes were recognized as up- and down-regulated. Because the identification of differentially-expressed genes are highly dependent on the transcriptomics pipeline, this proves that RNA-Seq techniques differ and it is essential to try to figure out the differences and similarities between each pipeline and DE programs and why we would use a certain one over another. 

Looking at the MA plots was helpful in determining the differences between the aligners used. If genes were shown to be further towards the 1 and -1 on the y-axis, this meant that they were more differentially expressed. The MA plot results tells us what the relative sensitivity of the differential expression programs. 

For all of the aligners, the abundance of the transcript varied with each DE program. When the Bowtie aligner was used with EdgeR and DESeq2, it seemed that Bowtie was more sensitive in its ability to represent the transcripts because there were more colored dots. When used with Limma, there were significantly fewer colored dots. Analyzing the Salmon aligner, we found that all of the genes were expressed at lower levels than compared to the other aligners. But, they were also spread out more across the x-axis. This means that they varied from low abundance to high abundance. The Sailfish aligner was the most sensitive because it identified the most number of genes expressed. The Kalliso aligner differed from the other aligners because it did not work well with Limma. It only identified 1 gene, while the other aligners identified at least 182.  

The differences in the results of these aligners may be due to their methodology when they are mapping gene sequences. Both Bowtie2 and Sailfish are splice-aware aligners (method = slow), so the sequence reads were mapped directly to the reference genome. Salmon and Kallisto, on the other hand, are pseudo-aligners (method = fast), meaning that the reads are not directly aligned. Instead, they are determined based on their compatability with known transcripts. 


a = abundance of transcript (counts per million)
genes on left = low abundance
genes on right = high abundance

m = fold change between the two samples
if a sample is on the top half, it is more highly expressed in the RNAi treated
if on the bo

color dots indicate genes that met the cut off, significiantly different
more dots = more sensitive, 

2. Differential Expression Programs


Looking at the three distinct differential expression programs we can see that there are notable patterns both in terms of the overall trends in total number of genes detected by each program and their specific detection patterns with each of the respective aligners. There is a concerningly large range in the number of detected differentially expressed genes both within single differential expression programs based on the chosen aligner and between the three separate DE programs. Certain differential expression program/aligner combinations immediately standout for their nearly complete lack of genes detected. For instance DESeq2 with Salmon only picked up two differentially expressed genes and Kallisto with Limma only found a single differentially expressed gene. In contrast Sailfish with DESeq2 reported over 600 differentially expressed genes and Salmon with edgeR resulted in over 450. Given such polar extremes it would seem likely that these could represent outliers where those with little to no detected differential genes are too conservsative and unreliable due to missing genes that were really differentially expressed whereas those with such high levels of reported differentially expressed genes are too leniant and resulting in false positives that make them unreliable as well. Some of the alligners did yield similar numbers of detected genes between differential expression programs. For instance the Salmon and Bowtie venn diagrams depict very similar numbers of identified genes in each DE program case.

Looking at each of the three differential expression program venn diagrams Limma has the lowest total number of reported genes making it an overall more conservative approach, but again there is high levels of variability in its relative level of conservativeness depending on which of the four alligners it was paired with. A consequence of the apparent more conservative threshold for Limma is that there are fewer genes that are common to three or more of the alligners when compared with DESeq2 and edgeR. Having high levels of genes that are commonly detected regardless of the selected alligner could be a marker of fidelity for that differential expression program and this could in turn suggest that DEseq2 and edgeR could have higher numbers of "true positives"" detected, meaning they capture more of the genes that were in actuality significantly differentially expressed. 

Overall these venn diagrams provide evidence for a general trend in which there is a tradeoff between conservativeness and true positive detection rate for the DE programs, but a limitation of this project is our lack of a definitive readout against which the DE programs and alligner/ DE program combinations could be compared against. The true test for the relative efficacy of each of the combinations would require an analysis of gene expression with RT-qPCR to develop a profile of the actual levels of each gene and then an assessment of how well each combination's set of detected differentially expressed genes match up to the RT-qPCR data to yield a ratio of specificity to true positive detection rate with each combination. Another study employed this type of comparison and concluded that the alligner had little weight in the overall quality of the analysis while the differential expression approach adopted was far more important. Under these conditions DESeq2 and Limma were found to have nearly identical true positive and specificity rates and these rates were the highest of the nine programs they tested. edgeR on the other hand was found to have an equallly high specificity, but a lower true positive rate (Costa-Silva 2017). Our results here likely stand in contrast to these findings due to the significantly lower rates of detected genes for limma compared to DeSeq2 which would make it seem rather difficult for them to share similar rates of both specificty and true positive rates. In addition, the lack of consistenccy between aligners seen in our data between the three differential expression programs would suggest that the choice of alligner can in fact have a more significant effect on the outcomes of the RNAseq pipeline than they found. However, since they only used some of the same aligners this is not a perfect comparison.  

In summary it is key to be careful when selecting both the alligner and the differential expressio program as there are some combinations that are likely to skew the results depending on their conservativeness and true detection rates. Our analysis here further demonstrates the vast importance of preregistration of the complete pipeline to avoid questionable research practices. One could dramatically alter the number of genes detected asimply by altering the pipeline at either the aligner or the expression program stage to manipulate the resutls in an irresponsible way. 

Source:
RNA-Seq differential expression analysis: An extended review and a software tool
Juliana Costa-Silva, Douglas Domingues, Fabricio Martins Lopes
https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0190152