---
title: "Project 3 Methods"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is a draft for the methods of Project 3 of BIOL 364 at Bucknell University. 
This project utilized 12 pipeline to detect differentlially expressed genes (Table). The pipeline was two parts, with the first part being the use of one of four aligners. Aligners were used through Bucknell University's connection to Galaxy, a tool use Bison Net, Bucknell's high performance computing cluster. This allowed use of aligners through a graphical user interface, rather than the command line. 

The second portion of the pipeline was the use of a program to detext differential expression of genes. Again, one of four programs were used (Table). This used a metadata file as well as a file of transciptomic data. These programs were also run though a graphical uner interface in DEBrowser, launched through the R package DEBrowser. Parameters used include: an adjusted p value of 0.01, two fold change. A Trimmed Mean of M-values (TMM) normalization was applied. The MA plots show all genes. 

Alligners:

  Kallisto

  bowtie2
  
  1) Use the RSEM-Bowtie2 tool
  2) Set `rsem reference` to "Build rsem reference"
  3) Set `reference fasta file` to the cdna.all.fa.gz file.
  4) Set `reference name` to something like drosophila_rsem
  5) Change `calculate expression with rsem` to "Just build rsem reference for latter rsem profiling"
  6) Execute!
  7) For each sample:
      1) Use the RSEM-Bowtie2 tool
      2) Set `rsem reference` to "rsem reference available from history"
      3) Set `Sample name` to the appropriate name for the sample (e.g. "S2_DRSC_Untreated-1")
      4) Set `Library type` to Single or Paired, as appropriate
      5) Set `FASTQ file` to the matching file(s)
      6) Set `Create bam results files` to "No BAM results files"
      7) Profit.

  Salmon

  Sailfish
    1) For `Select a reference transcriptome from your history or use a built-in index?` select "Use one from history".
    2) Select the Drosophila_melanogaster.BDGP6.cdna.all.fa.gz file for the reference genome
    3) Select Single-end and the FASTQ file(s) as appropriate
    4) Select the gtf.gz file for the `File containing a mapping of transcripts to genes`
    5) Leave all of the other options alone.
    6) Execute!


  
DEBrowser:
1) Load the Count Data File and the Metadata File
2) Filter the data using CPM, CPM<1, in at least 3 samples (half of the samples)
3) Batch correct the data using TMM normalization, Combat correction method, condition as Treatment, batch as Batch
4) Visualize the PCA plot by changing Color field to "condition", and Shape field to "batch"
5) Go to DE Analysis and Add New Comparison
   Set the Condition 1 samples to the Untreated and Condition 2 to the RNAi samples
   Chose the appropriate DE Method (edgeR, DESeq2, or lima), and leave the parameters on the default settings
   Go to Main Plots and explore the MA and Volcano plots

6) Confirm that RNAi experiment worked 

1) Load in the gene results and metadata
2) Filter and batch correct as above
3) Run the appropriate DE Analysis 
4) Go to the Tables and sort by log10padjust search for FBgn0261552 - this is the *pasilla* gene
  
  

