---
title: "Results MA Plots"
author: "Bio 364 Students"
date: "3/26/2019"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

#*Organized by aligner*


#Bowtie- Cheyenne

##limma
```{r}
bowtie_limma_MA <- knitr::include_graphics("figures/Bowtie_limma_MA_Moore.png") 
```

##DESeq
```{r}
bowtie_deseq_MA <- knitr::include_graphics("figures/bowtie2_deseq2_maplot.png") 
```

##EdgR
```{r}
bowtie_edger_MA <- knitr::include_graphics("figures/Bowtie2EdgeR_McGauley.png") 
```


##All Bowtie
```{r}
bowtie_limma_MA
bowtie_deseq_MA
bowtie_edger_MA

all_bowtie<- c(bowtie_limma_MA, bowtie_deseq_MA, bowtie_edger_MA)

all_bowtie
```
Based on initial viewing of the MA plots where Bowtie2 was the aligner used it looks like the number of differentially expressed genes varies with DE program (limma, DESeq, Edgr). Limma appears to have the least differentually expressed genes. DESeq and Edgr have more DE genes (about the same when comparing the two) but DESeq seems to have genes that are more differentially expressed, eg. further towards 1 and -1 on the y axis. 





#Kallisto - Jordan
##limma
##DESeq
##EdgR

```{r Kallisto MA plot figures} 
kallisto_limma_MA<- knitr::include_graphics("figures/Simpson.png") 
kallisto_edger_MA<- knitr::include_graphics("figures/MA_kallisto_edgeR.png")
kallisto_deseq_MA<- knitr::include_graphics("figures/kallistoDESeq_Sussman.png")
```

```{r Kallisto MA plot figures part 2}
kallisto_edger_MA
kallisto_limma_MA
kallisto_deseq_MA

all_kallisto_MA<- c(kallisto_edger_MA, kallisto_limma_MA, kallisto_deseq_MA)
```


#Salmon
##limma
##DESeq
##EdgeR

```{r salmon MA plot figures} 
salmon_limma_MA <- knitr::include_graphics("figures/McKinleySalmonLimmaMAPlot.png")
salmon_deseq_MA <- knitr::include_graphics("figures/salmon_deseq2_maplot.png") 
salmon_edger_MA <- knitr::include_graphics("figures/KE_Salmon_EdgeR_MAPlot.png") 

salmon_limma_MA
salmon_deseq_MA
salmon_edger_MA

all_salmon_MA <- c(salmon_limma_MA,salmon_deseq_MA,salmon_edger_MA)
```


#SailFish
##limma
##DESeq
##EdgR

```{r sailfish MA plot figures}
sailfish_edger_MA <- knitr::include_graphics("figures/JustinKoss.sailfish.edgeR.png") 
sailfish_limma_MA <- knitr::include_graphics("figures/Ritter_DEBrowser_image.png") 
sailfish_deseq_MA <- knitr::include_graphics("figures/Vogel_Sailfish_DESeq2_MAplot.png") 
sailfish_edger_MA
sailfish_limma_MA
sailfish_deseq_MA
```




#*Organized by DE Program*
bowtie
kallisto
salmon
sailfish

##limma
```{r limma DE program}
bowtie_limma_MA
kallisto_limma_MA
salmon_limma_MA
sailfish_limma_MA
```


##DESeq
```{r DESeq DE program}
bowtie_deseq_MA
kallisto_deseq_MA
salmon_deseq_MA
sailfish_deseq_MA
```

##EdgeR
```{r EdgeR DE program}
bowtie_edger_MA
kallisto_edger_MA
salmon_edger_MA
sailfish_edger_MA
```




#Possibly helpful links
https://stackoverflow.com/questions/3668102/plotting-several-jpeg-images-in-a-single-display
http://rforbiochemists.blogspot.com/2016/05/working-with-multiple-image-files.html